Test/MySafePriorityQueue.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace Test;
+
+public class MySafePriorityQueue<T, K> where K : IComparable<K>
+{
+    private SortedList<K, T> _list = new SortedList<K, T>();
+    private int _size;
+    private readonly object _locker = new();
+
+    public void Enqueue(T value, K priority)
+    {
+        lock (_locker)
+        {
+            _list.Add(priority, value);
+            _size++;
+            Monitor.Pulse(_locker);
+        }
+    }
+
+    public T Dequeue()
+    {
+        lock (_locker)
+        {
+            if (_list.Count == 0)
```

Comment:
```suggestion
            while (_list.Count == 0)
```
А если несколько потоков заблокируются на `Wait`? Затем в очередь добавят элемент, все эти потоки проснутся. Один из них возьмет элемент из очереди, а остальные упадут на `queue.Keys.Last()`. 
Я ведь говорил на паре, что не `if`, а `while` тут нужно использовать.

Reply:
Но я ведь не PulseAll пишу, а Pulse. Да и вроде все корректно работает в таком случае: 
![image](https://user-images.githubusercontent.com/64278750/202845444-e29b2255-03cf-490c-82e9-330f046e13c6.png)
Ни разу не упала программа в таком случае

Reply:
-1